src/instructlab/configuration.py

```diff
@@ -29,27 +30,81 @@
 # Local
 from . import log
 
+
+class STORAGE_DIR_NAMES:
+    ILAB = "instructlab"
+    DATASETS = "datasets"
+    CHECKPOINTS = "checkpoints"
+    MODELS = "models"
+    TAXONOMY = "taxonomy"
+    INTERNAL = (
+        "internal"  # for storing all ilab-internal files the user doesn't need to see
+    )
+
+
+# ILAB_CONFIG_HOME = os.path.join(USER_HOME_DIR, ".config", ".ilab")
+ILAB_CONFIG_HOME = None
+ILAB_DATA_HOME = None
+
+# default to "$HOME/.config" per the XDG standards
+if not os.getenv("XDG_CONFIG_HOME"):
+    if not os.getenv("HOME"):
+        raise RuntimeError(
+            "The HOME directory does not exist. Please make sure it's set and run this again."
+        )
+    ILAB_CONFIG_HOME = os.path.expandvars(
+        os.path.join("${HOME}", ".config", STORAGE_DIR_NAMES.ILAB)
+    )
+else:
+    ILAB_CONFIG_HOME = os.path.expandvars(
+        os.path.join("${XDG_CONFIG_HOME}", STORAGE_DIR_NAMES.ILAB)
+    )
+
+
+# figure out what to use for the data directory
+if not os.getenv("XDG_DATA_HOME"):
+    if not os.getenv("HOME"):
+        raise RuntimeError(
+            "The HOME directory does not exist. Please make sure it's set and run this again."
+        )
+    ILAB_DATA_HOME = os.path.expandvars(
+        os.path.join("${HOME}", ".local", "share", STORAGE_DIR_NAMES.ILAB)
+    )
+else:
+    ILAB_DATA_HOME = os.path.expandvars(
+        os.path.join("${XDG_DATA_HOME}", STORAGE_DIR_NAMES.ILAB)
+    )
+
```

Comment:
Minor nit (non-blocking) Some of the duplication and double lookups could be replaced with something like (need to scroll):

```suggestion
def getHomeDir(*subdir: str) -> str: 
    home = os.getenv("HOME")
    if not home:
        raise RuntimeError(
           "The HOME directory does not exist. Please make sure it's set and run this again."
        )
    return os.path.join(home, *subdir)  

def getHomeEnvPath(env: str, suffix: str, *homePaths: str) -> str:
    path = os.getenv(env) or getHomeDir(*homePaths)
    return os.path.join(path, suffix)

ILAB_CONFIG_HOME = getHomeEnvPath("XDG_CONFIG_HOME",  STORAGE_DIR_NAMES.ILAB, ".config")
ILAB_DATA_HOME =  getHomeEnvPath("XDG_DATA_HOME", STORAGE_DIR_NAMES.ILAB, ".local", "share")

```

Reply:
This is definitely cleaner than my original approach. I mainly wanted to avoid prematurely abstracting, but this describes it well. I'll include this in the PR.

Reply:
Thank you very much for this suggestion, I included it!